### PR TITLE
More external reader refinements

### PR DESCRIPTION
**Pull request by HT154**
* Rename `org.pkl.core.Readers` -> `org.pkl.core.Closeables`
* Remove coupling between `pkl.core.module`/`resource` and `pkl.core.messaging`

### CODE REVIEW

File: pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java
Comment (HT154) [inline pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java:1]: Not in love with the naming here, but it's necessary to disambiguate and accurately describes the functionality. These classes are a straight move, though git doesn't show that.

**Comment by translatenix** [comment]
I'd love to rein in the cyclic package dependencies between messaging, externalreader, resource, and module.
Is there time for me to investigate?

**Comment by bioball** [comment]
> I'd love to rein in the cyclic package dependencies between messaging, externalreader, resource, and module.
Is there time for me to investigate before the 0.27 release?

We are planning on doing a release on Monday. When you think you can get a PR in for this?

**Comment by translatenix** [comment]
> We are planning on doing a release on Monday. When you think you can get a PR in for this?

I can send a PR (or give up) by tonight. Probably makes sense to first merge this PR.

File: pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import org.pkl.core.SecurityManager;
+import org.pkl.core.SecurityManagerException;
+import org.pkl.core.messaging.Messages.ListModulesRequest;
+import org.pkl.core.messaging.Messages.ListModulesResponse;
+import org.pkl.core.messaging.Messages.ReadModuleRequest;
+import org.pkl.core.messaging.Messages.ReadModuleResponse;
+import org.pkl.core.module.ExternalModuleResolver;
+import org.pkl.core.module.PathElement;
+
+public class MessageTransportModuleResolver implements ExternalModuleResolver {
```
Comment (bioball) [inline pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java:36]: I don't think this needed to be moved? Package `externalreader` was not exporting `MessageTransport`.

Comment (HT154) [reply pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java]: I think I get it? I'm still tripping over how Java does things.

The idea here is to ensure that nothing in `org.pkl.core.messaging` ends up in the `org.pkl.core.module`/`resource` public API, right? Does that include the constructor for the resolver classes? Or am I missing something here?

Comment (bioball) [reply pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java]: Yeah; none of signatures of the public methods (including constructors) were using types from `org.pkl.core.messaging `, unless I missed something.

It's totally fine for private methods use to `org.pkl.core.messaging`.

**Review by bioball — APPROVED** [review]
LGTM!

File: pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java (hunk 1 above)
Comment (HT154) [reply pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java]: The constructors for the resolver classes take a `MessageTransport` so I these do need to move?

Comment (bioball) [reply pkl-core/src/main/java/org/pkl/core/messaging/MessageTransportModuleResolver.java]: Ah! My bad!

I just merged this PR. Can you re-submit the other change as a new PR?

**Comment by bioball** [comment]
> I can send a PR (or give up) by tonight. Probably makes sense to first merge this PR.

Sounds good!

**Pull request merged**